onboarding/service.go

```diff
@@ -53,26 +77,29 @@ func NewOnboardService(svc *Service, as influxdb.AuthorizationService, opts ...O
 }
 
 // IsOnboarding determine if onboarding request is allowed.
-func (s *OnboardService) IsOnboarding(ctx context.Context) (bool, error) {
+func (s *service) IsOnboarding(ctx context.Context) (bool, error) {
 	if s.alwaysAllow {
 		return true, nil
 	}
 
-	allowed := false
-	err := s.service.store.View(ctx, func(tx kv.Tx) error {
-		// we are allowed to onboard a user if we have no users or orgs
-		users, _ := s.service.store.ListUsers(ctx, tx, influxdb.FindOptions{Limit: 1})
-		orgs, _ := s.service.store.ListOrgs(ctx, tx, influxdb.FindOptions{Limit: 1})
-		if len(users) == 0 && len(orgs) == 0 {
-			allowed = true
-		}
-		return nil
-	})
-	return allowed, err
+	s.tenantService.RLock()
+	defer s.tenantService.RUnlock()
```

Comment:
It's potentially overkill to take the read-lock here, but I wanted to do it to check it was possible to make the types work. I could delete it.

Reply:
Since the lock will be given up when this function returns it doesn't seem like locking here will do much good. I suppose there's a race condition between returning this answer for onboarding being allowed and when the onboarding is complete - maybe somehow a different user could be created between those calls - but getting a lock over the entire onboarding operation is probably not worth the trouble in my opinion. Especially since it would need to be unlocked to actually do the onboarding I'd imagine...

Reply:
The lock here was meant to guard against a change between the calls to `FindUsers` and `FindOrganizations`, but agreed that as-is it's not very useful.